src/app/services/configuration-control.service.ts

```diff
@@ -0,0 +1,517 @@
+            key: key,
+            type: 'script',
+            label: this.setDisplayName(element),
+            description: element.description,
+            value: element.value,
+            readonly: element.readonly,
+            mandatory: element.mandatory,
+            order: element.order,
+            file: element.file,
+            editorOptions: this.setEditorConfig(key),
+            validity: element.validity
+          }));
+          break;
+        case 'CODE':
+          configurations.push(new CodeConfig({
+            key: key,
+            type: 'code',
+            label: this.setDisplayName(element),
+            description: element.description,
+            value: element.value,
+            readonly: element.readonly,
+            mandatory: element.mandatory,
+            order: element.order,
+            editorOptions: this.setEditorConfig(key),
+            validity: element.validity
+          }));
+          break;
+        case 'IPV4':
+        case 'IPV6':
+        default:
+          configurations.push(new TextboxConfig({
+            key: key,
+            type: 'text',
+            label: this.setDisplayName(element),
+            description: element.description,
+            value: element.value,
+            readonly: element.readonly,
+            mandatory: element.mandatory,
+            order: element.order,
+            validity: element.validity
+          }));
+          break;
+      }
+    });
+    return configurations.sort((a, b) => a.order - b.order);
+  }
+
+  public setEditorConfig(type: string) {
+    const editorOptions = {
+      mode: 'text/x-python',
+      lineNumbers: true,
+      lineWrapping: true,
+      foldGutter: true,
+      gutters: ['CodeMirror-linenumbers', 'CodeMirror-foldgutter', 'CodeMirror-lint-markers'],
+      autoCloseBrackets: true,
+      matchBrackets: true,
+      lint: true,
+      inputStyle: 'textarea'
+    };
+    if (type === 'JSON') {
+      editorOptions.mode = 'application/json';
+    }
+    return editorOptions;
+  }
+
+  public setDisplayName(configItem: any) {
+    if (configItem.hasOwnProperty('displayName')) {
+      return configItem.displayName.trim().length > 0 ? configItem.displayName : configItem.key;
+    }
+    return configItem.key;
+  }
+
+  /**
+   *
+   * @param pluginConfiguration plugin configuration
+   * @param groupConfigurations  group item configuration
+   * @returns
+   */
+  toFormGroup(pluginConfiguration: any, groupConfigurations: ConfigurationBase<string>[]) {
+    const group: any = {};
+    groupConfigurations.forEach(configuration => {
+      group[configuration.key] = configuration.required ?
+        new FormControl({ value: configuration.value || '', disabled: this.validateConfigItem(pluginConfiguration, configuration) }, Validators.required)
+        : new FormControl({ value: configuration.value || '', disabled: this.validateConfigItem(pluginConfiguration, configuration) });
```

Comment:
instead of this, should not we use 2nd arg as` , configuration.required ? Validators.required : null) `